#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t

gen = ParameterGenerator()

gen.add("max_planning_dist", double_t, 0, "Maximum distance the planner will look ahead.", 0.3, 0, 100.0)
gen.add("goal_accuracy", double_t, 0, "Maximal distance to the goal position.", 0.1, 0, 1.0)
gen.add("velocity", double_t, 0, "Maximum velocity the robot will drive.", 0.5, 0, 5.0)
gen.add("look_ahead_dist", double_t, 0, "Lookahead distance used by pure pursuit.", 0.1, 0, 1.0)
gen.add("look_ahead_ratio", double_t, 0, "Factor of the velocity the planner adds to lookahead distance.", 0.1, 0, 1.0)
gen.add("epsilon", double_t, 0, "Delta pose error threshold.", 0.000001, 0, 0.1)

exit(gen.generate("pure_pursuit_local_planner", "pure_pursuit_local_planner", "PurePursuitReconfigure"))